lib/testing/src/lib/process-services-cloud/pages/edit-process-filter-cloud-component.page.ts

```diff
@@ -27,14 +27,19 @@ export class EditProcessFilterCloudComponentPage {
     saveButton: ElementFinder = element(by.css('button[data-automation-id="adf-filter-action-save"]'));
     saveAsButton: ElementFinder = element(by.css('button[data-automation-id="adf-filter-action-saveAs"]'));
     deleteButton: ElementFinder = element(by.css('button[data-automation-id="adf-filter-action-delete"]'));
+    filter: ElementFinder = element(by.css(`adf-cloud-edit-process-filter mat-expansion-panel-header`));
 
     editProcessFilterDialogPage = new EditProcessFilterDialogPage();
 
     editProcessFilterDialog(): EditProcessFilterDialogPage {
         return this.editProcessFilterDialogPage;
     }
 
-    async clickCustomiseFilterHeader(): Promise<void> {
+    async isFilterDisplayed(): Promise<Boolean> {
```

Comment:
I'm a bit concerned about this type of usage.
method set to return Promise<Boolean>
but inside return waitUntilElementIsPresent, which returns Promise<any>, and which will never ever return false.

Reply:
Could you be more specific?
Async functions return promises, what I think you meant is not to wait for the promise to be resolved:
return BrowserVisibility.waitUntilElementIsPresent(this.filter);

Reply:
sorry, formatting removed parts of my message.
Edited now, makes a bit of sense?

Reply:
IMO, a promise of boolean method should return either _true_ or _false_.
While this method would return true in the happy case, and would timeout otherwise.

I think I saw in another PR of yours a different approach that I liked. Something like - writing from memory:
```
try
{
 await BrowserVisibility.waitUntilElementIsVisible(...);
 return true;
} catch {
 return false;
}
```

Reply:
do we need really to return true or false in this kind of cases?

Reply:
Depends on the usage. If it's to be used in an expect, then returning true or false makes sense.

Alternative would be to have promise of void as return type and remove "return"

I'd vote for boolean though, to be able to correctly use those types of methods in expects

Reply:
In this particular case, the method is used in an assertion.
If the element is visible, then it will return true, otherwise it might return false (falseIfMissing()) or time out.
If false is returned, then the assertion fails, otherwise the test will fail with timeout.
The artifice with try catch, was just for thoroughness, but I don't see it as a necessary need.